Assets/_BForBoss/_Core/Scripts/Enemies/FloatingTargetBehaviour.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using Perigon.Entities;
+using Perigon.Weapons;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class FloatingTargetBehaviour : EnemyBehaviour
+    {
+        [SerializeField] private HealthbarViewBehaviour _healthbar;
+        private AgentNavigationBehaviour _navigationBehaviour = null;
+        private EnemyShootingBehaviour _shootingBehaviour = null;
+        private FloatingTargetState _state = FloatingTargetState.MoveTowardsDestination;
+        
+        public void Initialize(Func<Vector3> getPlayerPosition, BulletSpawner bulletSpawner, Action onDeathCallback, Action<EnemyBehaviour> onReleaseToSpawner = null)
+        {
+            base.Initialize(getPlayerPosition, onDeathCallback, onReleaseToSpawner);
+            
+            _navigationBehaviour = GetComponent<AgentNavigationBehaviour>();
+            _navigationBehaviour.Initialize(getPlayerPosition, () =>
+            {
+                _state = FloatingTargetState.ShootTarget;
+                _shootingBehaviour.Reset();
+            });
+            
+            if (_healthbar != null)
+            {
+                _healthbar.Initialize(_lifeCycle);
+            }
+
+            _shootingBehaviour = GetComponent<EnemyShootingBehaviour>();
+            _shootingBehaviour.Initialize(getPlayerPosition, bulletSpawner, () =>
+            {
+                _state = FloatingTargetState.MoveTowardsDestination;
+            });
+        }
+
+        public override void Reset()
+        {
+            base.Reset();
+            gameObject.SetActive(true);
+            _healthbar.Reset();
+        }
```

Comment:
This is okay for now, but if this Enemy was spawned by the spawner, would we not want the gameObject return to the spawner when `Reset()` is called

Reply:
Sure, but that should be `EnemyBehaviour` job. i should trust that it's already been handled for me in there.

Reply:
Fair enough, will look into handling this in the EnemyManager Refactor Task